docs/contracts/v4/concepts/01-v4-vs-v3.mdx

```diff
@@ -1,30 +1,68 @@
 ---
-title: v4-versus-v3
+title: V4 vs V3
 ---
 
-While Uniswap v4's underlying concetrated liquidity is the same as Uniswap v3, there are some key differences in the architecture
-and accounting.
+While Uniswap v4's underlying concentrated liquidity is the same as Uniswap v3,
+there are some key differences in the architecture and accounting.
 
 # Singleton Design
 
 ### Pool Creation
 
-(TODO: talk about singleton vs factory, state vs contract/V3Pool)
+V4: The singleton contract facilitates the creation of a pool and 
+also stores its state. This pattern reduces costs when creating a pool
+and doing multi-hop swaps.
+
+V3: A factory contract is responsible for pool creation. The pool is
+a separate contract instance that manages its own state. Pool initialization
+is costly because of how gas consuming contract creation is.
 
 ### Flash Accounting
 
-(TODO: talk about flash accounting, multi-hop swaps)
+V4: The singleton uses "flash accounting", meaning a caller that unlocks it
+is allowed to freely access tokens given
+that prior to locking it back no tokens are owed to or from the caller. This
+further reduces the gas cost of
+trades that cross multiple pools and supports more complex
+integrations with Uniswap v4.
+
+V3: Because flash accounting is missing from V3, it is the responsibility
+of the swap router to execute swaps on each individual pool from which the
+multi-hop swap needs access to liquidity.
 
-# Liquidity Fee Accounting
+# Liquidity Fee Accounting 
 
-(TODO: talk about how fees must be collected on liquidity operation)
+V4: Accrued fees act like credit when modifying liquidity.
+Increasing liquidity will convert the fee revenue to liquidity
+inside the position while decreasing liquidity will automatically
+withdraw remaining fees.
 
-(TODO: talk about fees are automatically credited in increase liquidity)
+A parameter called "salt" can be provided when modifying liquidity. The
+salt can be used to have multiple positions in the same range.
+This separation may be preferred in hook enabled pools where the hook
+introduces further differentiations in liquidity provisioning. An example
+could be when out-of-range liquidity is repurposed and despite the fact that
+a user has two positions in the same range, each position is assigned to
+two different lending protocols in this case.
 
-(TODO: talk about salts)
+V3: Accrued fees are updated each time a liquidity operation occurs,
+however they don't account to liquidity deposit/withdrawal amounts 
+like in V4 and must be collected separately.
```

Comment:
```suggestion
**V3**: Liquidity positions of the same range and pool will share the same state. While believed to 
be more gas efficient at the time, integrating contracts will need to handle fee management since
the state is shared on the core pool contract
```